app/routes/task_routes.py

```diff
@@ -0,0 +1,162 @@
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, make_response, request, abort
+from datetime import datetime
+import requests, os
+
+task_list_bp = Blueprint("task_list_bp", __name__, url_prefix="/tasks")
+
+### Get all saved tasks or zero saved tasks
+### Sort Tasks By Title, Ascending
+### Sort Tasks By Title, Descending
+@task_list_bp.route("", methods=["GET"])
+def get_tasks():
+    # initialize list of tasks
+    tasks_response = []
+
+    # get sort param value
+    sort_query = request.args.get("sort")
+    
+    # sort tasks ascending if sort is asc
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+        
+    # sort tasks descending if sort is desc   
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    
+    # get all tasks if no param value
+    else:
+        tasks = Task.query.all()
+        
+    # loop thru each task      
+    for task in tasks:
+        # return task dict and add it to list
+        tasks_response.append(task.to_dict())
+
+    # return list of task dicts if they exist or empty list if no tasks 
+    return jsonify(tasks_response)
+
+### Create a Task: Valid Task With `null` `completed_at`
+### Create a Task: Invalid Task With Missing Data
+#### Missing `title`
+#### Missing `description`
+@task_list_bp.route("", methods=["POST"])
+def create_task():
+    # get request data
+    request_body = request.get_json()
+    # if title or description doesn't exist in request data, return 400 mesg
+    if "title" not in request_body or "description" not in request_body:
+        return {"details": "Invalid data"}, 400
+    
+    # create new task from request data
+    new_task = Task.from_dict(request_body)
+    
+    # add and commit new task
+    db.session.add(new_task)
+    db.session.commit()
+    
+    # return dict of task dicts
+    return {"task": new_task.to_dict()}, 201
+            
+### Get One Task
+@task_list_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    # return task if task_id valid and exists
+    task = validate_item(Task, task_id)
+    
+    # if goal doesn't exist, return task dict
+    if task.goal is None:
+            return {"task": task.to_dict() }
+    else:
+        # otherwise, return add goal_id:value pair and return
+        # dict of task dicts
+        task_with_goal = task.to_dict()
+        task_with_goal["goal_id"] = task.goal_id
+        return {"task": task_with_goal}
```

Comment:
```suggestion
    return_task = task.to_dict()
    if task.goal:
        return_task["goal_id"] = task.goal_id
    return {"task": return_task}
```
Small refactor to cut down on the number of lines. You could also update your `to_dict()` function to take care of conditionally adding the goal_id for you.